cmasher/utils.py

```diff
@@ -233,6 +239,102 @@ def _get_cmap_perceptual_rank(
 
 
 # %% FUNCTIONS
+# This function combines multiple colormaps at given nodes
+def combine_cmaps(
+    cmaps: list[Union[Colormap, str]],
+    nodes: Optional[list[float]] = None,
+    n_rgb_levels: int = 256,
+) -> LinearSegmentedColormap:
+    """Create a composite matplotlib colormap by combining multiple colormaps.
+
+    Parameters:
+        cmaps (list[Union[Colormap, str]]): list of colormaps to combine.
+        nodes (List[float], optional): The blending points between
+            colormaps, in the range [0, 1]. Defaults to equal divisions.
+        n_rgb_levels (int, optional): Number of RGB levels for each colormap
+            segment. Defaults to 256.
+
+    Returns:
+        Colormap: The composite colormap.
+
+    Raises:
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note:
+        The colormaps are combined from bottom to top.
+
+    References:
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
+
+    Examples:
+        # Example 1: Using colormap names
+        >>> cmaps = ["ocean", "prism", "coolwarm"]
+        >>> nodes = [0.2, 0.75]
+        >>> custom_cmap_1 = combine_cmaps(cmaps, nodes)
+
+        # Example 2: Using predefined colormaps
+        >>> cmap_0 = plt.get_cmap("Blues")
+        >>> cmap_1 = plt.get_cmap("Oranges")
+        >>> cmap_2 = plt.get_cmap("Greens")
+        >>> custom_cmap_2 = combine_cmaps([cmap_0, cmap_1, cmap_2])
+
+    """
+    # Check colormap datatype and convert to list[Colormap]
+    if len(cmaps) == 1:
+        raise ValueError("Cannot combine a single colormap.")
+    for cm in cmaps:
+        if not isinstance(cm, (Colormap, str)):
+            raise TypeError(f"Unsupported colormap type: {type(cm)}.")
+    _cmaps: list[Colormap] = [
+        cm if isinstance(cm, Colormap) else mpl.colormaps[cm] for cm in cmaps
+    ]
+
+    # Generate default nodes for equal separation
+    if nodes is None:
+        nodes = np.linspace(0, 1, len(_cmaps) + 1)[1:-1].tolist()
+
+    # Check nodes length
+    if len(nodes) != len(_cmaps) - 1:
+        raise ValueError(
+            "Number of nodes should be one less than the number of colormaps."
+        )
+
+    # Preprocess nodes list for easier segmentation
+    nodes.insert(0, 0.0)
+    nodes.append(1.0)
```

Comment:
I think a `np.ndarray` should be a valid input, and we could also allow unsorted inputs if we do the sorting ourselves
```suggestion
    nodes_arr = np.concatenate([[0], np.sort(nodes), [1]])
```

Reply:
Yes `np.ndarray` deservers a place here. 

But I intentionally didn't sort the `nodes` here because I think a unsorted `nodes` might be an indicator that there is a type or the users don't understand what they are doing. For example considering:
`custom_cmap = combine_cmaps(["ocean", "prism", "coolwarm"], nodes=[0.75, 0.25])`
They may intend to put the `node` between `"ocean"` and `""prism` at `0.75`, if we sort the list the result there would not be an error but the result would be unexpected (and user confused).

Or even worse, for the following:
`custom_cmap = combine_cmaps(["ocean", "prism", "coolwarm", "Blues"], nodes=[0.75, 0.25, 0.66])`
it's hard to explain what they intend to achieve but the code would still run without any error.

Maybe raise a warning in cases when `nodes` is not well ordered?

Reply:
You're right, sorting the input might be confusing, so let's not do it.